runtests.sh

```diff
@@ -553,6 +563,42 @@ then
 fi
 
 
+if [ $doRuffFormat = true ]
+then
+    set +e  # disable exit on failure so that diagnostics can be given on failure
+    if [ $doRuffFix = true ]
+    then
+        echo "${separator}${blue}ruff-fix${noColor}"
+    else
+        echo "${separator}${blue}ruff${noColor}"
+    fi
+
+    # ensure that the necessary packages for code format testing are installed
+    if ! is_pip_installed ruff
+    then
+        install_deps
+    fi
+    ${cmdPrefix}${PY_EXE} -m ruff --version
```

Comment:
getting error messages:
```
[2023-08-04T20:22:14.894Z] ruff
[2023-08-04T20:22:14.894Z] Traceback (most recent call last):
[2023-08-04T20:22:14.894Z]   File "/usr/lib/python3.8/runpy.py", line 194, in _run_module_as_main
[2023-08-04T20:22:14.894Z]     return _run_code(code, main_globals, None,
[2023-08-04T20:22:14.894Z]   File "/usr/lib/python3.8/runpy.py", line 87, in _run_code
[2023-08-04T20:22:14.894Z]     exec(code, run_globals)
[2023-08-04T20:22:14.894Z]   File "/usr/local/lib/python3.8/dist-packages/ruff/__main__.py", line 34, in <module>
[2023-08-04T20:22:14.894Z]     ruff = find_ruff_bin()
[2023-08-04T20:22:14.894Z]   File "/usr/local/lib/python3.8/dist-packages/ruff/__main__.py", line 30, in find_ruff_bin
[2023-08-04T20:22:14.894Z]     raise FileNotFoundError(path)
[2023-08-04T20:22:14.894Z] FileNotFoundError: /root/.local/bin/ruff
[2023-08-04T20:22:14.894Z] Traceback (most recent call last):
[2023-08-04T20:22:14.894Z]   File "/usr/lib/python3.8/runpy.py", line 194, in _run_module_as_main
[2023-08-04T20:22:14.894Z]     return _run_code(code, main_globals, None,
[2023-08-04T20:22:14.894Z]   File "/usr/lib/python3.8/runpy.py", line 87, in _run_code
[2023-08-04T20:22:14.894Z]     exec(code, run_globals)
[2023-08-04T20:22:14.894Z]   File "/usr/local/lib/python3.8/dist-packages/ruff/__main__.py", line 34, in <module>
[2023-08-04T20:22:14.894Z]     ruff = find_ruff_bin()
[2023-08-04T20:22:14.894Z]   File "/usr/local/lib/python3.8/dist-packages/ruff/__main__.py", line 30, in find_ruff_bin
[2023-08-04T20:22:14.894Z]     raise FileNotFoundError(path)
[2023-08-04T20:22:14.894Z] FileNotFoundError: /root/.local/bin/ruff
[2023-08-04T20:22:14.894Z] Check failed!
```
I'll try to use `ruff` directly as per suggested https://github.com/astral-sh/ruff/issues/2520#issuecomment-1416102075

Reply:
Hi @wyli  Following Ruff author's suggestion is good. Could you please provide the result of `which is ruff` and `whereis -b ruff` in this error environment? I would like to confirm if there is any non-default binary installation path in the testing environment since current Ruff should cover default binary installation paths correctly by:
```python
import sysconfig
print(sysconfig.get_path("scripts"))
print(sysconfig.get_path("scripts", scheme="posix_user"))
```

Reply:
it's 
```bash

[2023-08-05T16:42:12.532Z] ruff
[2023-08-05T16:42:12.532Z] /usr/local/bin/ruff
[2023-08-05T16:42:12.532Z] ruff: /usr/local/bin/ruff
```
since the commands on the dev branch now work fine for the testing pipelines, perhaps `python -m ruff` is not a critical issue

Reply:
Thanks for providing. Ruff is installed to a correct path in this env. That error might be caused by `sudo` or permission issues. Using `ruff` directly is good.